src/playground/components/CodeEditor.js

```diff
@@ -10,33 +10,72 @@ import {
 } from "../utils/codemirror-theme";
 import "../scss/editor.scss";
 
-export default function CodeEditor({
-	codeValue,
-	onUpdate,
-	eslintOptions,
-	eslintInstance,
-}) {
-	const extensions = useMemo(
-		() => [
-			history(),
-			bracketMatching(),
-			linter(esLint(eslintInstance, eslintOptions), { delay: 0 }),
-			javascript(),
-			ESLintPlaygroundTheme,
-			ESLintPlaygroundHighlightStyle,
-		],
-		[eslintOptions, eslintInstance],
-	);
-
-	return (
-		<CodeMirror
-			value={codeValue}
-			minWidth="100%"
-			height="100%"
-			extensions={extensions}
-			onChange={value => {
-				onUpdate(value);
-			}}
-		/>
-	);
-}
+const CodeEditor = forwardRef(
+	({ codeValue, onUpdate, eslintOptions, eslintInstance }, ref) => {
+		const editorRef = useRef(null);
+
+		const extensions = useMemo(
+			() => [
+				history(),
+				bracketMatching(),
+				linter(esLint(eslintInstance, eslintOptions), { delay: 0 }),
+				javascript(),
+				ESLintPlaygroundTheme,
+				ESLintPlaygroundHighlightStyle,
+			],
+			[eslintOptions, eslintInstance],
+		);
+
+		useImperativeHandle(ref, () => ({
+			scrollToPosition(line, col) {
+				if (editorRef.current) {
+					const editorView = editorRef.current.view;
+					if (editorView) {
+						const { state } = editorView;
+						const pos = state.doc.line(line).from + col;
+
+						// Set the cursor selection to the position
+						editorView.dispatch({
+							selection: { anchor: pos },
+							scrollIntoView: true,
+						});
+
+						const linePos = editorView.coordsAtPos(
+							state.doc.line(line).from,
+						);
+
+						// Calculate to try to center the line in the editor
+						if (linePos) {
+							const editorRect =
+								editorView.dom.getBoundingClientRect();
+							const offset =
+								linePos.top -
+								editorRect.top -
+								editorRect.height / 2;
+							editorView.scrollDOM.scrollTop += offset;
+						}
+
+						editorView.focus();
+					}
+				}
```

Comment:
```suggestion
                const editorView = editorRef.current?.view;
                if (editorView) {
                    const { state } = editorView;
                    const pos = state.doc.line(line).from + col;

                    // Set the cursor selection to the position
                    editorView.dispatch({
                        selection: { anchor: pos },
                        scrollIntoView: true,
                    });

                    const linePos = editorView.coordsAtPos(
                        state.doc.line(line).from,
                    );

                    // Calculate to try to center the line in the editor
                    if (linePos) {
                        const editorRect =
                            editorView.dom.getBoundingClientRect();
                        const offset =
                            linePos.top -
                            editorRect.top -
                            editorRect.height / 2;
                        editorView.scrollDOM.scrollTop += offset;
                    }

                    editorView.focus();
                }
```

Removed Unnecessary Null Checks for editorRef.current and editorRef.current.view